sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangePartitionKeyImpl.java

```diff
@@ -30,73 +37,125 @@ public PartitionKeyInternal getPartitionKeyInternal() {
     }
 
     @Override
-    public void accept(FeedRangeVisitor visitor) {
-        checkNotNull(visitor, "Argument 'visitor' must not be null");
-        visitor.visit(this);
-    }
-
-    @Override
-    public <TInput> void accept(GenericFeedRangeVisitor<TInput> visitor, TInput input) {
-        checkNotNull(visitor, "Argument 'visitor' must not be null");
-        visitor.visit(this, input);
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        FeedRangePartitionKeyImpl that = (FeedRangePartitionKeyImpl)o;
+        return Objects.equals(this.partitionKey, that.partitionKey);
     }
 
     @Override
-    public <T> Mono<T> accept(FeedRangeAsyncVisitor<T> visitor) {
-        checkNotNull(visitor, "Argument 'visitor' must not be null");
-        return visitor.visit(this);
+    public int hashCode() {
+        return Objects.hash(partitionKey);
     }
 
     @Override
-    public Mono<UnmodifiableList<Range<String>>> getEffectiveRanges(
+    public Mono<Range<String>> getEffectiveRange(
         IRoutingMapProvider routingMapProvider,
-        String containerRid,
-        PartitionKeyDefinition partitionKeyDefinition) {
+        RxDocumentServiceRequest request,
+        Mono<Utils.ValueHolder<DocumentCollection>> collectionResolutionMono) {
+
+        checkNotNull(
+            collectionResolutionMono,
+            "Argument 'collectionResolutionMono' must not be null");
 
-        String effectivePartitionKey = this.partitionKey.getEffectivePartitionKeyString(
-            this.partitionKey,
-            partitionKeyDefinition);
-        Range<String> range = Range.getPointRange(effectivePartitionKey);
-        ArrayList<Range<String>> rangeList = new ArrayList<>();
-        rangeList.add(range);
+        return collectionResolutionMono
+            .flatMap(documentCollectionResourceResponse -> {
 
-        return Mono.just((UnmodifiableList<Range<String>>)UnmodifiableList.unmodifiableList(rangeList));
+                final DocumentCollection collection = documentCollectionResourceResponse.v;
+                if (collection == null) {
+                    throw new IllegalStateException("Collection cannot be null");
+                }
+
+                final String effectivePartitionKey =
+                    this.partitionKey.getEffectivePartitionKeyString(
+                    this.partitionKey,
+                    collection.getPartitionKey());
+
+                Range<String> range = Range.getPointRange(effectivePartitionKey);
+                return Mono.just(range);
+            });
     }
 
     @Override
-    public Mono<UnmodifiableList<String>> getPartitionKeyRanges(
+    public Mono<List<String>> getPartitionKeyRanges(
```

Comment:
given that this is a FeedRange for a partition key, is the returned List always a List of size 1?

Reply:
Yes - only for the Epk overload it could contain multiple ranges.